3party/opening_hours/parse_months.cpp

```diff
@@ -31,17 +31,17 @@ namespace osmoh
 
       date_offset = ((lit('+')[_a = true] | lit('-')[_a = false])
                      >> charset::no_case[wdays] >> day_offset)
-      [bind(&DateOffset::SetWDayOffset, _val, _1),
-       bind(&DateOffset::SetOffset, _val, _2),
-       bind(&DateOffset::SetWDayOffsetPositive, _val, _a)]
+      [(bind(&DateOffset::SetWDayOffset, _val, _1),
+        bind(&DateOffset::SetOffset, _val, _2),
+        bind(&DateOffset::SetWDayOffsetPositive, _val, _a))]
```

Comment:
What is the logic behind this? Why additional _tuple_ here?

Reply:
It's not a tuple. Top-level comma expression in array subscript is deprecated. It means that we can't write
```cpp
arr[a, b, c]
```

In C++20 this expression is deprecated because in C++23 it is a multidimensional operator [] - https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2128r6.pdf
```cpp
// 2 x 3 array of arrays
array aa {array{1, 2, 3}, array{4, 5, 6}};
assert( (aa[0,1] == 2) );
assert( (aa[1,2] == 6) );
```

So, the fix for this warning is to cover it with `(...)`
```cpp
arr[(a, b, c)]
```